### PR TITLE
INFRA: Implement Provisioning Project

### PR DESCRIPTION
**Pull request by mimendel**
Closes Task #39590609 - INFRA: Implement Provisioning Project

### CODE REVIEW

File: DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.Plans.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public async ValueTask<IAppServicePlan> CreatePlanAsync(
+            string planName,
+            IResourceGroup resourceGroup)
+        {
+            IWithPricingTier appServicePlanWithPricing = this.azure.AppServices.AppServicePlans
+                .Define(planName)
+                    .WithRegion(Region.USWest)
+                        .WithExistingResourceGroup(resourceGroup);
+            return await appServicePlanWithPricing.WithPricingTier(PricingTier.StandardS1)
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.Plans.cs:23]: Newline break before return statements.

File: DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ResourceGroups.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ResourceGroups.cs:1]: Missing copyrights header

File: DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/ICloudBroker.Storages.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.Azure.Management.Sql.Fluent;
+
+namespace DMX.Core.Api.Infrastructure.Provision.Brokers.Clouds
+{
+    internal partial interface ICloudBroker
+    {
+        ValueTask<ISqlServer> CreateSqlServerAsync(
+            string sqlServerName,
+            IResourceGroup resourceGroup);
+        ValueTask<ISqlDatabase> CreateSqlDatabaseAsync(
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/ICloudBroker.Storages.cs:17]: Newline break required between multi-line statements

File: DMX.Core.Api.Infrastructure.Provision/Brokers/Configurations/ConfigurationBroker.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Brokers/Configurations/ConfigurationBroker.cs:1]: Missing copyrights header

File: DMX.Core.Api.Infrastructure.Provision/Models/Storages/SqlDatabase.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Models/Storages/SqlDatabase.cs:4]: Remove the extra line

File: DMX.Core.Api.Infrastructure.Provision/Program.cs [hunk 6]
```diff
@@ -0,0 +1,3 @@
+﻿using System;
+
+Console.WriteLine("Hello World!");
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Program.cs:3]: Shouldn't this actually call the service to provision?

File: DMX.Core.Api.Infrastructure.Provision/Services/Foundations/CloudManagements/ICloudManagementService.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        ValueTask<ISqlServer> ProvisionSqlServerAsync(
+            string projectName,
+            string environment,
+            IResourceGroup resourceGroup);
+
+        ValueTask<SqlDatabase> ProvisionSqlDatabaseAsync(
+            string projectName,
+            string environment,
+            ISqlServer sqlServer);
+        ValueTask<IWebApp> ProvisionWebAppAsync(
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Services/Foundations/CloudManagements/ICloudManagementService.cs:33]: Newline break required

**Comment by AndrewMcClelland** [comment]
Pulling latest from `main` will resolve build gate failure

**Review by AndrewMcClelland — CHANGES_REQUESTED** [review]
Pulling latest from `main` will resolve build gate failure

File: DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ResourceGroups.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        public async ValueTask<bool> CheckResourceGroupExistsAsync(string resourceGroupName) =>
+            await this.azure.ResourceGroups.ContainAsync(resourceGroupName);
+
+        public async ValueTask<IResourceGroup> CreateResourceGroupAsync(
+            string resourceGroupName)
+        {
+            return await this.azure.ResourceGroups
+                .Define(resourceGroupName)
+                .WithRegion(Region.USWest)
```
Comment (ShriHumrudha) [inline DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ResourceGroups.cs:22]: Paremeter Aliasing

Comment (AndrewMcClelland) [reply DMX.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ResourceGroups.cs:21]: The passed in values match the parameter names in part and remains readable without aliases, so don't think they're necessary here.

File: DMX.Core.Api.Infrastructure.Provision/Services/Processings/CloudManagements/CloudManagementProcessingService.cs [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            this.cloudManagementService = new CloudManagementService();
+        }
+
+        public async ValueTask ProcessAsync()
+        {
+            CloudManagementConfiguration configuration = this.configurationBroker.GetConfiguration();
+            await ProvisionResourcesAsync(configuration);
+            await DeprovisionResourcesAsync(configuration);
+        }
+        private async Task ProvisionResourcesAsync(CloudManagementConfiguration configuration)
```
Comment (hassanhabib) [inline DMX.Core.Api.Infrastructure.Provision/Services/Processings/CloudManagements/CloudManagementProcessingService.cs:34]: new line here

**Review by hassanhabib — APPROVED** [review]

**Pull request merged**